### PR TITLE
Set up rake task to create paypal plans and update db

### LINKED ISSUE #3220: Create rake task for setting up paypal api integration
We need to create the plans in the paypal server to use them later to create and execute agreements off them

### PR DESCRIPTION
**Pull request by mattwr18**
- Co-authored-by: Mike Aono <aonomike@gmail.com>
- Co-authored-by: Cire DIA <ciremoussadia@gmail.com>

#### Issue addressed

fixes #3220

### CODE REVIEW

File: app/services/paypal_service.rb [hunk 1]
```diff
@@ -7,7 +7,22 @@
...
+  def initialize(plan)
+    @plan = plan
+  end
+
+  def create_and_activate_recurring_plan
+    plan = if  @plan.third_party_identifier === 'premium'
+             PayPal::SDK::REST::Plan.new(plan_params_with_trial)
+           else
+             PayPal::SDK::REST::Plan.new(plan_params_without_trial)
+           end
```
Comment (tansaku) [inline app/services/paypal_service.rb]: alternatively could be 

```rb
plan = PayPal::SDK::REST::Plan.new(@plan.third_party_identifier === 'premium' ? plan_params_with_trial : plan_params_without_trial)
```

or

```rb
plan_type = @plan.third_party_identifier === 'premium' ? plan_params_with_trial : plan_params_without_trial
plan = PayPal::SDK::REST::Plan.new(play_type)
```

which make the method a little shorter, but let me know what you think about whether it's actually an improvement in readability?

File: app/services/paypal_service.rb [hunk 2]
```diff
@@ -22,6 +37,92 @@ def self.execute_agreement(agreement_token)
...
+  def activate_plan_params
+    {
+      op: 'replace',
+      value: { state: 'ACTIVE' },
+      path: '/'
+    }
+  end
+
+  def plan_params_with_trial
+    {
```
Comment (tansaku) [inline app/services/paypal_service.rb]: one possibility to get round the codeclimate issues here could be to get these JSON things into erb templates? but maybe that should be a refactoring ticket after we have this out?

**Comment by tansaku** [comment]
great work here @mattwr18 @ciremoussadia and @aonomike - I added a couple of minor comments/suggestions, but nothing blocking here I think :-)

File: app/services/paypal_service.rb (hunk 2 above)
Comment (aonomike) [reply app/services/paypal_service.rb]: @mattwr18 @tansaku I am not certain I understand how to implement this suggestion. do we create the erb under views/subscription directory? How do I pass the instance variables from service to the erb? maybe its a lil vague for me.

Comment (tansaku) [reply app/services/paypal_service.rb]: does this help? https://www.stuartellis.name/articles/erb/

the idea is that you have a template called my_json.erb and you evaluate it - same as with erb on other places you set instance variables ...

**Comment by tansaku** [comment]
great work @mattwr18 @aonomike and @ciremoussadia ! :-)

**Pull request merged**